Fix iOS build for OpenSSL.

olliwang opened the pull request:
The iOS target names has been changed for a long time and `-sysroot` is required to avoid build errors.

<!--- Use this part of template if you're adding new package. Remove the rest. -->
<!--- BEGIN -->

* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/create/cmake.html)
  step by step carefully. **[Yes|No]**

<!--- Remove next line if there is no corresponding "New package" issue. -->
* My change will resolve this "New package" request: https://github.com/cpp-pm/hunter/issues/<number>

---
<!--- END -->

<!--- Use this part of template if you're updating existing package. Remove the rest. -->
<!--- BEGIN -->

* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/update.html)
  step by step carefully. **Yes**

---
<!--- END -->

<!--- Use this part of template for other type of changes. Remove the rest. -->
<!--- BEGIN -->

* I've checked this [Git style guide](https://0.readthedocs.io/en/latest/git.html). **Yes**
* I've checked this [CMake style guide](https://0.readthedocs.io/en/latest/cmake.html). **Yes**
* My change will work with CMake 3.2 (minimum requirement for Hunter). **Yes**
* I will try to keep this pull request as small as possible and will try not to mix unrelated features. **Yes**

---
<!--- END -->

rbsheth reviewed (commented):
Hi @olliwang, are you sure this works? CI for iOS is failing

olliwang commented:
> Hi @olliwang, are you sure this works? CI for iOS is failing

Absolutely. I tested this with latest Xcode 12.4, iOS SDK 14.4 on macOS Big Sur 11.1.

My iOS project used Hunter v0.23.254 and the OpenSSL built without error. However, after I updated the Hunter version I saw errors like this:

```
cd /Users/username/.hunter/_Base/017e688/b1345bb/31f95b6/Build/OpenSSL/Source/arm64 && ./Configure iphoneos-cross threads no-shared -miphoneos-version-min=9.0 "" no-async --prefix=/Users/username/.hunter/_Base/017e688/b1345bb/31f95b6/Build/OpenSSL/Install "-arch arm64"

Failure!  build file wasn't produced.
Please read INSTALL and associated NOTES files.  You may also have to look over
your available compiler tool chain or change your configuration.

target already defined - iphoneos-cross (offending arg: )
make: *** [/Users/username/.hunter/_Base/017e688/b1345bb/31f95b6/Build/OpenSSL/Build/OpenSSL-arm64-prefix/src/OpenSSL-arm64-stamp/Debug-iphoneos/OpenSSL-arm64-configure] Error 255
Command /bin/sh failed with exit code 2

** BUILD FAILED **
```

The `target already defined - iphoneos-cross (offending arg: )` error was caused by the empty ${no-asm} so there is no character after `offending arg:`.

Then I built again and got errors like this:

```
Performing configure step for 'OpenSSL-x86_64'
cd /Users/username/.hunter/_Base/017e688/b1345bb/31f95b6/Build/OpenSSL/Source/x86_64 && ./Configure threads no-shared -miphoneos-version-min=9.0 no-asm 1 no-async --prefix=/Users/username/.hunter/_Base/017e688/b1345bb/31f95b6/Build/OpenSSL/Install "-arch x86_64"
Usage: Configure [no-<cipher> ...] [enable-<cipher> ...] [-Dxxx] [-lxxx] [-Lxxx] [-fxxx] [-Kxxx] [no-hw-xxx|no-hw] [[no-]threads] [[no-]shared] [[no-]zlib|zlib-dynamic] [no-asm] [no-egd] [sctp] [386] [--prefix=DIR] [--openssldir=OPENSSLDIR] [--with-xxx[=vvv]] [--config=FILE] os/compiler[:flags]

pick os/compiler from:
BS2000-OSD BSD-generic32 BSD-generic64 BSD-ia64 BSD-sparc64 BSD-sparcv8 
BSD-x86 BSD-x86-elf BSD-x86_64 Cygwin Cygwin-i386 Cygwin-i486 Cygwin-i586 
Cygwin-i686 Cygwin-x86 Cygwin-x86_64 DJGPP MPE/iX-gcc UEFI UWIN VC-CE VC-WIN32 
VC-WIN32-ARM VC-WIN32-ONECORE VC-WIN64-ARM VC-WIN64A VC-WIN64A-ONECORE 
VC-WIN64A-masm VC-WIN64I aix-cc aix-gcc aix64-cc aix64-gcc android-arm 
android-arm64 android-armeabi android-mips android-mips64 android-x86 
android-x86_64 android64 android64-aarch64 android64-mips64 android64-x86_64 
bsdi-elf-gcc cc darwin-i386-cc darwin-ppc-cc darwin64-ppc-cc 
darwin64-x86_64-cc gcc haiku-x86 haiku-x86_64 hpux-ia64-cc hpux-ia64-gcc 
hpux-parisc-cc hpux-parisc-gcc hpux-parisc1_1-cc hpux-parisc1_1-gcc 
hpux64-ia64-cc hpux64-ia64-gcc hpux64-parisc2-cc hpux64-parisc2-gcc hurd-x86 
ios-cross ios-xcrun ios64-cross ios64-xcrun iossimulator-xcrun iphoneos-cross 
irix-mips3-cc irix-mips3-gcc irix64-mips4-cc irix64-mips4-gcc linux-aarch64 
linux-alpha-gcc linux-aout linux-arm64ilp32 linux-armv4 linux-c64xplus 
linux-elf linux-generic32 linux-generic64 linux-ia64 linux-mips32 linux-mips64 
linux-ppc linux-ppc64 linux-ppc64le linux-sparcv8 linux-sparcv9 linux-x32 
linux-x86 linux-x86-clang linux-x86_64 linux-x86_64-clang linux32-s390x 
linux64-mips64 linux64-s390x linux64-sparcv9 mingw mingw64 nextstep 
nextstep3.3 sco5-cc sco5-gcc solaris-sparcv7-cc solaris-sparcv7-gcc 
solaris-sparcv8-cc solaris-sparcv8-gcc solaris-sparcv9-cc solaris-sparcv9-gcc 
solaris-x86-gcc solaris64-sparcv9-cc solaris64-sparcv9-gcc solaris64-x86_64-cc 
solaris64-x86_64-gcc tru64-alpha-cc tru64-alpha-gcc uClinux-dist 
uClinux-dist64 unixware-2.0 unixware-2.1 unixware-7 unixware-7-gcc vms-alpha 
vms-alpha-p32 vms-alpha-p64 vms-ia64 vms-ia64-p32 vms-ia64-p64 vos-gcc 
vxworks-mips vxworks-ppc405 vxworks-ppc60x vxworks-ppc750 vxworks-ppc750-debug 
vxworks-ppc860 vxworks-ppcgen vxworks-simlinux 

NOTE: If in doubt, on Unix-ish systems use './config'.
Configuring OpenSSL version 1.1.1d (0x1010104fL) for 1
Using os-specific seed configuration
make: *** [/Users/username/.hunter/_Base/017e688/b1345bb/31f95b6/Build/OpenSSL/Build/OpenSSL-x86_64-prefix/src/OpenSSL-x86_64-stamp/Debug-iphoneos/OpenSSL-x86_64-configure] Error 1
Command /bin/sh failed with exit code 2

** BUILD FAILED **
```

It told me the target name was not invalid, and after searching on OpenSSL's GitHub. I realized `iphoneos-cross` was a legacy target name and should be removed long time ago. After fixing the target names, I got errors like this:

```
echo "No update step for 'OpenSSL-x86_64'"
No update step for 'OpenSSL-x86_64'
/usr/local/Cellar/cmake/3.19.3/bin/cmake -E echo_append
/usr/local/Cellar/cmake/3.19.3/bin/cmake -E touch /Users/username/.hunter/_Base/017e688/b1345bb/31f95b6/Build/OpenSSL/Build/OpenSSL-x86_64-prefix/src/OpenSSL-x86_64-stamp/Debug-iphoneos/OpenSSL-x86_64-update
echo "No patch step for 'OpenSSL-x86_64'"
No patch step for 'OpenSSL-x86_64'
/usr/local/Cellar/cmake/3.19.3/bin/cmake -E echo_append
/usr/local/Cellar/cmake/3.19.3/bin/cmake -E touch /Users/username/.hunter/_Base/017e688/b1345bb/31f95b6/Build/OpenSSL/Build/OpenSSL-x86_64-prefix/src/OpenSSL-x86_64-stamp/Debug-iphoneos/OpenSSL-x86_64-patch
echo "Performing configure step for 'OpenSSL-x86_64'"
Performing configure step for 'OpenSSL-x86_64'
cd /Users/username/.hunter/_Base/017e688/b1345bb/31f95b6/Build/OpenSSL/Source/x86_64 && ./Configure darwin64-x86_64-cc threads no-shared -miphoneos-version-min=9.0 no-asm "" no-async --prefix=/Users/username/.hunter/_Base/017e688/b1345bb/31f95b6/Build/OpenSSL/Install "-arch x86_64"

Failure!  build file wasn't produced.
Please read INSTALL and associated NOTES files.  You may also have to look over
your available compiler tool chain or change your configuration.

target already defined - darwin64-x86_64-cc (offending arg: )
make: *** [/Users/username/.hunter/_Base/017e688/b1345bb/31f95b6/Build/OpenSSL/Build/OpenSSL-x86_64-prefix/src/OpenSSL-x86_64-stamp/Debug-iphoneos/OpenSSL-x86_64-configure] Error 255
Command /bin/sh failed with exit code 2

** BUILD FAILED **
```

It seems it was building the library for iPhone Simulator (x86_64) but using the iOS SDK (arm64). So I added `-sysroot` explicitly and it finally compiled without any issue.

rbsheth reviewed (approved):
Sounds good, thank you!

rbsheth commented:
Released: https://github.com/cpp-pm/hunter/releases/tag/v0.23.294
